File: app/src/main/java/com/nurullo/foodsharing/model/requestParams/RegisterParams.java
package com.nurullo.foodsharing.model.requestParams;

import androidx.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class RegisterParams {
    @Expose
    @SerializedName("username")
    private final String username;

    @Expose
    @SerializedName("email")
    private final String email;

    @Expose
    @SerializedName("phone")
    private final String phone;

    @Expose
    @SerializedName("nickname")
    private final String nickname;

    @Expose
    @SerializedName("password")
    private final String password;

    public RegisterParams(String username, String email, String phone, String nickname, String password) {
        this.username = username;
        this.email = email;
        this.phone = phone;
        this.nickname = nickname;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public String getEmail() {
        return email;
    }

    public String getPhone() {
        return phone;
    }

    public String getNickname() {
        return nickname;
    }

    public String getPassword() {
        return password;
    }
}

File: app/src/main/java/com/nurullo/foodsharing/model/requestParams/RegisterParams.kt
package com.nurullo.foodsharing.model.requestParams

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.nurullo.foodsharing.model.response.FoodAdPojo

class RegisterParams     constructor (username: kotlin.String, email: kotlin.String, phone: kotlin.String, nickname: kotlin.String, password: kotlin.String){
    @Expose @SerializedName("username")
    val  username: kotlin.String

    @Expose @SerializedName("email")
    val  email: kotlin.String

    @Expose @SerializedName("phone")
    val  phone: kotlin.String

    @Expose @SerializedName("nickname")
    val  nickname: kotlin.String

    @Expose @SerializedName("password")
    val  password: kotlin.String

    @Expose @SerializedName("foodAds")
    private  val  foodAds: kotlin.collections.MutableList<FoodAdPojo> = java.util.ArrayList<FoodAdPojo>()

    init {
        this.username = username
        this.email = email
        this.phone = phone
        this.nickname = nickname
        this.password = password
        foodAds.add(
            FoodAdPojo("", "", "", "", "", 0L,
            "", "", -2L, "", "",
            0.0, 0.0, false, 0L)
        )
    }
}
